src/main/java/com/iridium/iridiumskyblock/commands/CommandManager.java

```diff
@@ -6,74 +6,123 @@
         commands.add(command);
     }
 
-    public void unRegisterCommand(Command command) {
+    /**
+     * Unregisters a single command in the command system.
+     * @param command The command which should be registered
+     */
+    public void unregisterCommand(Command command) {
         commands.remove(command);
     }
 
+    /**
+     * Method which handles command execution for all sub-commands.
+     * Automatically checks if a User can execute the command.
+     * All parameters are provided by Bukkit.
+     *
+     * @param commandSender The sender which executes this command
+     * @param cmd The Bukkit {@link org.bukkit.command.Command} representation
+     * @param label The label of this command. Not used.
+     * @param args The arguments of this command
+     * @return true if this command was executed successfully
+     */
     @Override
-    public boolean onCommand(CommandSender cs, org.bukkit.command.Command cmd, String s, String[] args) {
-        if (args.length != 0) {
-            for (Command command : commands) {
-                if (command.aliases.contains(args[0]) && command.enabled) {
-                    if (command.player && !(cs instanceof Player)) {
-                        // Must be a player
-                        cs.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().mustBeAPlayer
-                                .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
-                        return true;
-                    }
-                    if ((cs.hasPermission(command.permission) || command.permission
-                            .equalsIgnoreCase("") || command.permission
-                            .equalsIgnoreCase("iridiumskyblock.")) && command.enabled) {
-                        command.execute(cs, args);
-                    } else {
-                        // No permission
-                        cs.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().noPermission
-                                .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
-                    }
-                    return true;
-                }
-            }
-        } else {
-            if (cs instanceof Player) {
-                Player p = (Player) cs;
+    public boolean onCommand(@NotNull CommandSender commandSender, org.bukkit.command.@NotNull Command cmd, @NotNull String label, String[] args) {
+        if (args.length == 0) {
+            if (commandSender instanceof Player) {
+                Player p = (Player) commandSender;
+                // TODO: Send message
                 return true;
             }
         }
-        cs.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().unknownCommand
+
+        for (Command command : commands) {
+            // We don't want to execute other commands or ones that are disabled
+            if (!(command.aliases.contains(args[0]) && command.enabled)) {
```

Comment:
this shouldnt be casesensitive

Reply:
Then it was not case sensitive in the first place, I just applied the early return patern

Reply:
oh it wasnt... well it was supposed to be lol. Ill take a look at that later